roseus/roseus.cpp

```diff
@@ -1548,6 +1548,44 @@ pointer ROSEUS_DELETE_PARAM(register context *ctx,int n,pointer *argv)
   return((ros::param::del(key))?(T):(NIL));
 }
 
+pointer ROSEUS_SEARCH_PARAM(register context *ctx,int n,pointer *argv)
+{
+  string key, result;
+
+  ckarg(1);
+  if (isstring(argv[0])) key.assign((char *)get_string(argv[0]));
+  else error(E_NOSTRING);
+
+  if ( ros::param::search(key, result) ) {
+    return makestring((char *)result.c_str(), result.length());
+  }
+  return(NIL);
+}
+
+pointer ROSEUS_LIST_PARAM(register context *ctx,int n,pointer *argv)
+{
+  ckarg(0);
+
+#if ROS_VERSION_MINIMUM(1,11,17)
+  std::vector<std::string> keys;
+  if ( ros::param::getParamNames(keys) ) {
+    pointer ret = cons(ctx, NIL,NIL), first;
+    first = ret;
+    vpush(ret);
+    for(std::vector<std::string>::iterator it = keys.begin(); it != keys.end(); it++) {
+      const std::string& key = *it;
+      ccdr(ret) = cons(ctx, makestring((char *)key.c_str(), key.length()), NIL);
+      ret = ccdr(ret);
+    }
+    vpop();
+    return ccdr(first);
+  }
+#else
+  ROS_WARN("%s : ros::rosparam::getParamNames is not impremented for roscpp %d.%d.%d",  __PRETTY_FUNCTION__, ROS_VERSION_MAJOR, ROS_VERSION_MINOR, ROS_VERSION_PATCH);
```

Comment:
ROS_ERROR?
Also, typo in _implemented_

Reply:
thank you, use ROS_ERROR and fixed typo.